digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/MassTransitStopService.scala

```diff
@@ -164,19 +230,31 @@ trait MassTransitStopService extends PointAssetOperations {
         updateMunicipality(id, municipalityCode)
         updateAssetGeometry(id, point)
       }
-      getPersistedStopWithPropertiesAndPublishEvent(id, municipalityCode, geometry)
+      getPersistedStopWithPropertiesAndPublishEvent(id, { _ => roadLink })
     }
   }
 
   def updateExistingById(id: Long, optionalPosition: Option[Position], properties: Set[SimpleProperty], username: String, municipalityValidation: Int => Unit): MassTransitStopWithProperties = {
     updateExisting(withId(id), optionalPosition, properties, username, municipalityValidation)
   }
 
-  private def fetchRoadLink(linkId: Long): Option[(Int, Seq[Point])] = {
-    vvhClient.fetchVVHRoadlink(linkId).map{ x => (x.municipalityCode, x.geometry) }
+  def updateAdministrationClassValue(assetId: Long, administrativeClass: AdministrativeClass): Unit ={
+    massTransitStopDao.updateNumberPropertyValue(assetId, "linkin_hallinnollinen_luokka", administrativeClass.value)
+  }
+
+  private def updateFloatingReasonValue(assetId: Long, floatingReason: FloatingReason): Unit ={
+    massTransitStopDao.updateNumberPropertyValue(assetId, "kellumisen_syy", floatingReason.value)
+  }
+
+  private def deleteFloatingReasonValue(assetId: Long): Unit ={
+    massTransitStopDao.deleteNumberPropertyValue(assetId, "kellumisen_syy")
+  }
+
+  private def fetchRoadLink(linkId: Long): Option[VVHRoadlink] = {
+    vvhClient.fetchVVHRoadlink(linkId)
   }
 
-  override def create(asset: NewMassTransitStop, username: String, geometry: Seq[Point], municipality: Int): Long = {
+  override def create(asset: NewMassTransitStop, username: String, geometry: Seq[Point], municipality: Int, roadlink: Option[VVHRoadlink]): Long = {
```

Comment:
Would it be enough to pass just administrative class value here instead of the whole road link?